command/agent.go

```diff
@@ -466,6 +466,13 @@ func (cmd *AgentCommand) readConfig() *agent.Config {
 		cmd.UI.Error(fmt.Sprintf("Failed to parse config: %v", err))
 		return nil
 	}
+	// More post processing of the config
```

Comment:
I think the post-processing of the configuration shouldn't be done in `command/agent.go` but in the config code so that this is transparent to all usages of the configuration. Having said that I think we should leave it here for now to resolve the issue. I'm going to touch this next and I can pick this up if you want.